app/controllers/foreman_tasks/api/recurring_logics_controller.rb

```diff
@@ -38,10 +38,22 @@ def cancel
       def resource_class
         ForemanTasks::RecurringLogic
       end
+                      
+      api :POST, '/recurring_logics/bulk/destroy', N_('Delete recurring logics by search query')
+      param :search, String, :desc => N_('Search query'), :required => true
+      def bulk_destroy
+        scope = resource_scope.search_for(params[:search])
+        destroyed = scope.destroy_all
+        if destroyed
```

Comment:
`destroy_all` returns a collection of objects that were destroyed and calling it on an empty scope returns empty array, so the condition will always be true

Reply:
changed to `destroy!` to get the error, and `destroyed` is now just the scope